docs/policy.md

```diff
@@ -0,0 +1,169 @@
+Same goes for `"buffer.Buffer": true` in builtins - when the package does `const buffer = require('buffer')` it will get an object with just one reference to `Buffer` constructor.
+
+LavaMoat can often generate precise lists of fields used by the package. You can narrow them down further if need be by creating a `policy-override.json` file next to the `policy.json`.
+
+# Using policy-override to expand or limit access
+
+## Adding access where LavaMoat's policy was too strict
+
+Policy files are generated by statically analyzing the source of packages and finding out which powerful references they seem to be using. It's impossible to detect it 100% correctly because there are JavaScript constructs that can't be detected without running the code.  
+If such missing permissions are identified, it's best to doublecheck they're expected to be there and then add them to the policy-override file.
+
+Example:  
+Allow `entropoetry` access to the global `URL` class and expand access to the entire `buffer` builtin on top of the generated policy above.
+
+`./lavamoat/node/policy-override.json`:
+```
+{
+  "resources": {
+    "some-package>entropoetry": {
+      "globals": {
+        "URL": true,
+      }
+    },
+    "some-package>entropoetry>bn.js": {
+      "builtin": {
+        "buffer": true
+      }
+    }
+  }
+}
+```
+
+## Limiting access
+
+If LavaMoat's static analysis finds references used in a package, but in a functionality that's not called in your application it's reasonable to want them denied. 
+
+Example:  
+- Forbid requiring of `assert` by `entropoetry`
+- Deny access to `console` with the exception of `console.log`
+
+`./lavamoat/node/policy-override.json`:
+```
+{
+  "resources": {
+    "some-package>entropoetry": {
+      "builtin": {
+        "assert": false
+      },
+      "globals": {
+        "console": false,
+        "console.log": true
+      }
+    }
+  }
+}
+```
+
+### ⚠️ Unsupported deny-list style policy
+
+> **Warning**  
+> Explicitly allowing access to an entire item and then attempting to narrow it down is not supported. The order of items in the JSON file should not matter.
+> As a result the following will not work as intuitively expected:
+
+```
+/*** broken example ***/
+  "globals": {
+    "a": true,
+    "a.b": false
+  }
+/*** broken example ***/
+```    
+
+`a.b` will remain accessible since `a` was declared as accessible in its entirety.
+
+To properly deny access to `a.b` while allowing access to other fields of `a` you'd need an allow-list style policy like this:
+```
+  "globals": {
+    "a.c": true,
+    "a.d": true
+  }
+```
+
+----
+
+More specifically, the deepest chain of properties needs to point to an allowed reference and skipping all-but-necessary access on higher levels is possible, so the following will work and allow `a.z` and `a.b.c` but not `a.b.y`
```

Comment:
I'm not finding this explanation to square with the previous example. Does this example actually work, i.e. does `"a.b": false` get applied here?

Reply:
Yes, because while a is allowed and a.b.c is allowed, we can skip allowing the rest of a.b along with c. 

This is a super narrow edge case and it's not been used anywhere yet. (See note at the bottom)